File: scripts/fetal_scripts/process_covariates.jl
#!/usr/env/bin julia

using CSV,DataFrames

processcov = function(covfilename, qualfilename, quantfilename)
    cov = CSV.read(covfilename, DataFrame,
                   transpose = true)
    cov[:,:famid] .= 0
    qualcov = cov[:, unique(vcat("famid", "id",
                                  names(cov, AbstractString)
                                 ))
                  ]
    quantcov = cov[:, unique(vcat("famid", "id",
                                   names(cov, Number)
                                  ))
                   ]

    CSV.write(qualfilename,
              qualcov;
              delim='\t',
              header=false
             )

    CSV.write(quantfilename,
              quantcov;
              delim='\t',
              header=false
             )
    return
end

processcov(snakemake.input[1], snakemake.output[1], snakemake.output[2])

# processcov(
#             "/u/project/gandalm/cindywen/isoform_twas/eqtl_new/data/eur/chuanjiao/tri1_25HCP.txt",
#             "fin_RNA_seq/Trimester_1_EUR_50_HPC_gene")
# 
# processcov(
#             "/u/project/gandalm/cindywen/isoform_twas/eqtl_new/data/eur/chuanjiao/tri2_15HCP.txt",
#             "fin_RNA_seq/Trimester_2_EUR_50_HPC_gene")

File: scripts/fetal_scripts/process_covariates.R
library(data.table)
cov = read.table('/u/project/gandalm/cindywen/isoform_twas/eqtl_new/data/eur/50HCP_cov_trimester1.txt',
                 header=FALSE,
                 sep='\t'
)

t_cov = transpose(cov[,-1]) # Omit covariate name

t_cov = cbind(rep(x=0, times=dim(t_cov)[1]), t_cov)
colnames(t_cov) = c('fam_id', cov[,1])

# Generate covariates
qual_cov=t_cov[,c("fam_id", "id", "sex")]

# Generate qcovariates
quant_cov=t_cov[,-8]

write.table(qual_cov, 'fin_RNA_seq/Trimester_1_EUR_50_HPC_gene/qual_covars.txt',
            quote=FALSE,
            sep='\t',
            col.names=FALSE,
            row.names=FALSE
)
write.table(quant_cov, 'fin_RNA_seq/Trimester_1_EUR_50_HPC_gene/quant_covars.txt',
            quote=FALSE,
            sep='\t',
            col.names=FALSE,
            row.names=FALSE
)

cov = read.table('/u/project/gandalm/cindywen/isoform_twas/eqtl_new/data/eur/50HCP_cov_trimester2.txt',
                 header=FALSE,
                 sep='\t'
)

t_cov = transpose(cov[,-1]) # Omit covariate name

t_cov = cbind(rep(x=0, times=dim(t_cov)[1]), t_cov)
colnames(t_cov) = c('fam_id', cov[,1])

# Generate covariates
qual_cov=t_cov[,c("fam_id", "id", "sex")]

# Generate qcovariates
quant_cov=t_cov[,-8]

write.table(qual_cov, 'fin_RNA_seq/Trimester_2_EUR_50_HPC_gene/qual_covars.txt',
            quote=FALSE,
            sep='\t',
            col.names=FALSE,
            row.names=FALSE
)
write.table(quant_cov, 'fin_RNA_seq/Trimester_2_EUR_50_HPC_gene/quant_covars.txt',
            quote=FALSE,
            sep='\t',
            col.names=FALSE,
            row.names=FALSE
)
